### PR TITLE
feat: pre-push hook

### PR DESCRIPTION
**Pull request by eugypalu**
# Pull Request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build-related changes
- [ ] Documentation content changes
- [ ] Testing

# What is the new behavior?

Adds a git hook that runs the following checks during the pre-push phase:
```
cargo check --all-features
cargo clippy --workspace --all-features --all-targets -- -D warnings
cargo fmt --all --check
```
This helps to catch any issues before the CI process.

To add the hook to `.git/hooks`, run the script `scripts/install_hooks.sh` or run `make setup`.

# Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: .hooks/pre-push [hunk 1]
```diff
@@ -0,0 +1,21 @@
+#!/bin/sh
+
+cargo check
```
Comment (greged93) [inline .hooks/pre-push:3]:
```suggestion
cargo check --all-features
```

File: scripts/install_hooks.sh [hunk 2]
```diff
@@ -0,0 +1,4 @@
+cp .hooks/pre-push .git/hooks/pre-push
+chmod +x .git/hooks/pre-push
+
+echo "Git hooks installed successfully."
```
Comment (greged93) [inline scripts/install_hooks.sh:4]: missing a line break here

**Comment by tcoratger** [comment]
@eugypalu In the PR description, you added some instructions to install the hooks, maybe we could add this to the contributing section of the README for contributors to be aware of this, what do you think?

File: .hooks/pre-push [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+    exit 1
+fi
+
+cargo clippy --workspace --all-features --all-targets -- -D warnings
+if [ $? -ne 0 ]; then
+    echo "Pre-push hook failed: cargo clippy detected warnings or errors."
+    exit 1
+fi
+
+cargo fmt --all --check
```
Comment (tcoratger) [inline .hooks/pre-push:15]: I think we should put `cargo +nightly fmt --all --check` no? because of fmt is based on nightly no?

Comment (eugypalu) [reply .hooks/pre-push]: Fixed. I had always used it without nightly, but it makes sense to me to add it. Thanks

**Comment by eugypalu** [comment]
> @eugypalu In the PR description, you added some instructions to install the hooks, maybe we could add this to the contributing section of the README for contributors to be aware of this, what do you think?

README updated, thanks.

**Review by greged93 — APPROVED** [review]

**Pull request merged**